zopkio/test_runner.py

```diff
@@ -317,7 +322,10 @@ def _execute_single_test(self, config, failure_handler, naarad_obj, test):
         # 2 ways of loop 1. loop each test (Default) or 2.loop after the entire suite
         while (test.current_iteration < test.total_number_iterations):
           test.current_iteration = test.current_iteration + 1
-          self._run_and_verify_test(test)
+          #verify if the test has previouslt failed. If so then don't try to run again
+          #unless the config complete_all_iterations is set
+          if ((test.result != constants.FAILED) or runtime.get_active_config("complete_all_iterations",False)):
```

Comment:
I recommend a integer config (maybe consecutive_failures_per_test) for finer granularity. Unfortunately will make this a bit more difficult as you need to keep track of additional state (but its not really too much more)